src/Symfony/Component/DependencyInjection/Tests/Fixtures/containers/container_almost_circular.php

```diff
@@ -102,6 +103,34 @@
 $container->register('subscriber2', 'stdClass')->setPublic(false)
     ->addArgument(new Reference('manager2'));
 
+// doctrine-like event system with listener
+
+$container->register('manager3', 'stdClass')
+    ->setLazy(true)
+    ->setPublic(true) // just for assertions
+    ->addArgument(new Reference('connection3'));
+
+$container->register('connection3', 'stdClass')
+    ->setProperty('listener', [new Reference('listener3')]);
+
+$container->register('listener3', 'stdClass')
+    ->setPublic(true) // just for assertions
+    ->setProperty('manager', new Reference('manager3'));
+
+// doctrine-like event system with small differences
+
+$container->register('manager4', 'stdClass')
```

Comment:
This is a second case that's actually unrelated, but I stumbled upon it. For some reason, if `connection4` is NOT public, you get a nesting exception.

Reply:
Could be worth to add this explanation as a comment code ?